libs/store/src/lib/effects/player.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  AppIdNumberType,
+  AppPlayerId,
+  AppPlayerLog,
+  AppPlayerLogId,
+  AppPlayerState,
+  AppRaceId,
+  AppTournamentId,
+  AppTournamentState,
+} from '@razor/models';
+import { generateAvatarLink, generateUid } from '@razor/util';
+import {
+  invalidPlayerName,
+  invalidPlayerNameLength,
+  playerNotFound,
+  tournamentNotFound,
+} from '../loggers';
+import { raceNotFound } from '../loggers/race';
+import {
+  clearPlayerPayload,
+  joinPlayerPayload,
+  sendTypeLogPlayload,
+} from '../payloads';
+import { Dispatch, RootState } from '../store';
+
+export const joinPlayer = async (
+  dispatch: Dispatch,
+  payload: joinPlayerPayload,
+  state: RootState,
+): Promise<void> => {
+  const { tid, playerName }: { tid: string; playerName: string } = payload;
+  let tournamentId: string;
+
+  if (playerName.length < 2 || playerName.length > 10)
```

Comment:
Why are we not terminating the the execution of the effect when the constraints fail?

Reply:
Fix in PR #14 under commit 2c90ad8bcbf7254f6da140cca1193fcea40da26e